dali/server/daldap.cpp

```diff
@@ -164,23 +166,28 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
 
                 CDateTime now;
                 now.setNow();
-                if (now.compare(reqUTCTimestamp, false) < 0)//timestamp from the future?
+                CDateTime daliTime(now);
+                if (requestSignatureAllowedClockVarianceSeconds)//allow for clock variance between machines
+                    daliTime.adjustTimeSecs(requestSignatureAllowedClockVarianceSeconds);
+
+                if (daliTime.compare(reqUTCTimestamp, false) < 0)//timestamp from the future?
                 {
                     StringBuffer localDaliTimeUTC;
-                    now.getString(localDaliTimeUTC, false);//get UTC timestamp
-                    ERRLOG("LDAP: getPermissions(%s) scope=%s user=%s Request digital signature UTC timestamp %s from the future (Dali UTC time %s)",key?key:"NULL",obj?obj:"NULL",username.str(), requestTimestamp.str(), localDaliTimeUTC.str());
+                    daliTime.getString(localDaliTimeUTC, false);//get UTC timestamp
+                    ERRLOG("LDAP: getPermissions(%s) scope=%s user=%s Request digital signature UTC timestamp %s from the future (Dali adjusted UTC time %s). Check configured allowedClockVariance (%d sec)",key?key:"NULL",obj?obj:"NULL",username.str(), requestTimestamp.str(), localDaliTimeUTC.str(), requestSignatureAllowedClockVarianceSeconds);
                     return SecAccess_None;//deny
                 }
 
-                CDateTime expiry;
-                expiry.set(now);
+                CDateTime expiry(now);
                 expiry.adjustTime(requestSignatureExpiryMinutes);//compute expiration timestamp
+                if (requestSignatureAllowedClockVarianceSeconds)//allow for clock variance between machines
+                    expiry.adjustTimeSecs(requestSignatureAllowedClockVarianceSeconds);
```

Comment:
Is the expiry meant to be the oldest allowed timestamp?  meaning now() - minutes until expire - allowed variance?

I think maybe? your logic is reversed.  you should be subtracting from now to make the oldest allowed time... and your comparison should be reversed

reqUTCTimestamp.compare(expiry, false) < 0

Reply:
Agree, fixed and tested